Snakefile

```diff
@@ -452,3 +452,26 @@ rule map:
     shell:
         "scripts/map.py {input.img} {input.labels} {output} {input.predicts}"
 
+### extract image rule: for recovering the source images given a set of labels
+rule extract_images:
+    """for recovering the source images given a set of labels from segment_map(_exp).tiff"""
+    input:
+        sourceDir =  config['out']+"/{sample}/rev_transforms/",
+    params:
+        labels =  config['extracted_labels'],
+    output:
+        config['out']+"/{sample}/label_Images.txt"
+    conda: "envs/default.yml"
+    shell:
+        "scripts/extract_images.py {input.sourceDir} {params.labels} {output}"
```

Comment:
hey @tommyfuu,
does this work? in [my comment earlier](https://github.com/beelabhmc/flower_map/pull/17#issuecomment-829757968), I suggested switching the order of `{output}` and `{params.labels}` because I wasn't sure whether argparse would be able to figure out how many labels there are, if there end up being multiple of them
but I guess it could just use the last argument as the `output` and use everything between the first and last arguments as labels, even if it doesn't know how many labels the user will provide?

Reply:
ohhhh that's a good point! I just left it there and in my testing nothing is wrong. I can switch it back!

Reply:
ah! well if it didn't cause any trouble then feel free to ignore this!